hazelcast-sql/src/test/java/com/hazelcast/jet/sql/StaleIndexTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql;
+
+import com.hazelcast.config.IndexConfig;
+import com.hazelcast.config.IndexType;
+import com.hazelcast.map.IMap;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.Serializable;
+
+import static java.util.Collections.singletonList;
+
+public class StaleIndexTest extends SqlTestSupport {
+    private IMap<Integer, FooTuple> map;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        initialize(3, null);
+    }
+
+    @Test
+    public void test() {
+        map = instance().getMap("m");
+        createMapping("m", int.class, FooTuple.class);
+
+        createIndexAndExecuteQuery("f0");
+        map.destroy();
+        createIndexAndExecuteQuery("f1");
+
+        sleepSeconds(3); // plan should be invalidated now
```

Comment:
I do not like operating on time in our tests., maybe moving that test to ```com.hazelcast.sql.impl.plan.cache``` and make ```PlanCache#plans``` package private and wait for it to be cleared would be better?